### ISSUE TITLE
Gorouter frontend should allow configuration of idle timeout

### ISSUE THREAD

**Issue by staylor14** [cloudfoundry/gorouter#189]
## Issue
Welcome to the issue!  In today's issue, we cover how it would be awesome if the gorouter frontend had a configurable idle timeout.

## Context
It turns out that not every IaaS's load balancers work the same.  In particular, GCP's global load balancers requires some special handling.  In troubleshooting 502 errors coming from GCP's load balancers, we were directed to [this blog post](https://blog.percy.io/tuning-nginx-behind-google-cloud-platform-http-s-load-balancer-305982ddb340) by Google's support engineer.

TL;DR, Google's load balancers don't like it when the backends hang up on an idle connection in fewer than 650 seconds.

We're aware that other IaaS's load balancers want different behaviors, so we're asking that this be configurable by CF operators.

## Steps to Reproduce
1. Put CF on GCP with BBL
1. Set the gorouter frontend's idle timeout to 1 second (this is just to demonstrate the problem so a shorter timeout reproduces the problem faster).
1. Hit your CF api with the following script a lot:
```
#!/bin/bash

while true; do
  bash -c "cf auth <user> <password> &"
  sleep 0.2
done
```
If you're like us, you'll see 502 errors every 2 - 3 minutes on average.
## Expected result

I expect no 502s from the GCP global load balancers that say "backend_connection_closed_before_data_sent_to_client"

## Current result
Several 502s from the GCP global load balancers that say "backend_connection_closed_before_data_sent_to_client"

## Possible Fix
Make the gorouter's frontend idle timeout configurable, and on GCP, set it to 650s.

## `name of issue` Output Results
Que?

**Comment by ljfranklin** [comment]
CAPI team also sees this error on an almost daily basis in our CI pipelines. Allowing us to configure this property would save us a lot of pain going forward.

**Comment by shalako** [comment]
Prioritized https://www.pivotaltracker.com/story/show/150561986

**Comment by CAFxX** [comment]
Partially related to item 2 in https://github.com/cloudfoundry/gorouter/issues/181#issuecomment-318257511 (that is about read/write timeouts, this is about idle timeouts)

**Comment by kkallday** [comment]
I'm glad to see this is being addressed. On releng, we deploy `CF` many times a day and noticed random `502`.

**Comment by aaronshurley** [comment]
This feature was shipped with v0.163.0:  https://github.com/cloudfoundry-incubator/routing-release/releases/tag/0.163.0

Closing this issue. Please let us know if you have any further concerns.
